src/Microsoft.ML.TorchSharp/NasBert/TextClassificationTrainer.cs

```diff
@@ -0,0 +1,857 @@
+
+                Directory.CreateDirectory(destDir);
+
+                string relativeFilePath = Path.Combine(destDir, destFileName);
+
+                int timeout = 10 * 60 * 1000;
+                using (var ch = (_parent._host as IHostEnvironment).Start("Ensuring model file is present."))
+                {
+                    var ensureModel = ResourceManagerUtils.Instance.EnsureResourceAsync(_parent._host, ch, ModelUrl, destFileName, destDir, timeout);
+                    ensureModel.Wait();
+                    var errorResult = ResourceManagerUtils.GetErrorMessage(out var errorMessage, ensureModel.Result);
+                    if (errorResult != null)
+                    {
+                        var directory = Path.GetDirectoryName(errorResult.FileName);
+                        var name = Path.GetFileName(errorResult.FileName);
+                        throw ch.Except($"{errorMessage}\nmodel file could not be downloaded!");
+                    }
+                }
+
+                return relativeFilePath;
+            }
+
+            public void Validate(IProgressChannel pch, IChannel ch, int epoch)
+            {
+                var validationSet = _parent._options.ValidationSet;
+                Model.eval();
+
+                DataViewRowCursor cursor = default;
+                if (_parent._options.Sentence2ColumnName != default)
+                    cursor = validationSet.GetRowCursor(validationSet.Schema[_parent._options.Sentence1ColumnName], validationSet.Schema[_parent._options.Sentence2ColumnName], validationSet.Schema[_parent._options.LabelColumnName]);
+                else
+                    cursor = validationSet.GetRowCursor(validationSet.Schema[_parent._options.Sentence1ColumnName], validationSet.Schema[_parent._options.LabelColumnName]);
+
+                var sentence1Getter = cursor.GetGetter<ReadOnlyMemory<char>>(validationSet.Schema[_parent._options.Sentence1ColumnName]);
+                var sentence2Getter = _parent._options.Sentence2ColumnName != default ? cursor.GetGetter<ReadOnlyMemory<char>>(validationSet.Schema[_parent._options.Sentence2ColumnName]) : default;
+                var labelGetter = cursor.GetGetter<uint>(validationSet.Schema[_parent._options.LabelColumnName]);
+
+                // Pre-allocate the memory so it's only done once (though this step needs to be optimized)
+                List<Tensor> inputTensors = new List<Tensor>(_parent._options.BatchSize);
+                List<long> targets = new List<long>(_parent._options.BatchSize);
+                int numCorrect = 0;
+                int numRows = 0;
+
+                var cursorValid = true;
+                while (cursorValid)
+                {
+                    cursorValid = ValidateStep(cursor, sentence1Getter, sentence2Getter, labelGetter, ref inputTensors, ref targets, ref numCorrect, ref numRows);
+                }
+                Accuracy = numCorrect / (float)numRows;
```

Comment:
>Accuracy = numCorrect / (float)numRows; [](http://example.com/codeflow?start=16&length=39)

is it possible we canrun into case returing numRos is 0.0f and cause the Accuracy be infinity?